articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+    mssql@mssqlserver:/$ /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P 'Passw0rd!'
+    1> CREATE DATABASE todos_db
+    2> GO
+    1> exit
+    mssql@mssqlserver:/$ exit
+    ```
+
+### Run the demo application locally
+
+Follow the next steps to build and run the application locally.
+
+1. Build the Bootable JAR. When we are building the Bootable JAR, we need to specify the database driver version we want to use:
+ 
+    ```bash
+    todo-list (bootable-jar) $ MSSQLSERVER_DRIVER_VERSION=7.4.1.jre11 \
+    mvn clean package
+    ```
+
+1. Launch the Bootable JAR by using the following command. When we are launching the application, we need to pass the required environment variables to configure the data source:
+
+    ```bash
+    todo-list (bootable-jar) $ MSSQLSERVER_USER=SA \
+    MSSQLSERVER_PASSWORD=Passw0rd! \
+    MSSQLSERVER_JNDI=java:/comp/env/jdbc/mssqlds \
+    MSSQLSERVER_DATABASE=todos_db \
+    MSSQLSERVER_HOST=localhost \
+    MSSQLSERVER_PORT=1433 \
+    java -jar ./target/todo-list-bootable.jar -Djboss.node.name=node1
+    ```
+
+1. (Optional) If you want to verify the clustering capabilities, you can also launch more instances of the same application by changing the `jboss.node.name` argument and, to avoid conflicts with the port numbers, shifting the port numbers by using `jboss.socket.binding.port-offset` argument. For example, to launch a second instance that will represent a new pod on OpenShift, you can execute the following command in a new terminal window:
+    
+    ```bash  
+    todo-list (bootable-jar) $ MSSQLSERVER_USER=SA \
+    MSSQLSERVER_PASSWORD=Passw0rd! \
+    MSSQLSERVER_JNDI=java:/comp/env/jdbc/mssqlds \
+    MSSQLSERVER_DATABASE=todos_db \
+    MSSQLSERVER_HOST=localhost \
+    MSSQLSERVER_PORT=1433 \
+    java -jar ./target/todo-list-bootable.jar -Djboss.node.name=node2 -Djboss.socket.binding.port-offset=10
+    ```
+
+    > [!NOTE]
+    > By default the Bootable JAR configures the JGroups subsystem to use the UDP protocol and sends messages to discover other cluster members to the 230.0.0.4 multicast address. To properly verify the clustering capabilities on your local machine, your Operating System should be capable of sending and receiving multicast datagrams and route them to the 230.0.0.4 IP through your ethernet interface. Check your network configuration and verify this is working if you see warnings related to the cluster on the server logs.
+
+1. Open `http://localhost:8080/` in your browser to visit the application home page. If you have created more instances, you can access them by shifting the port number, for example `http://localhost:9080/`. The application will look similar to the following image:
+
+    ![ToDo EAP demo Application](./media/howto-deploy-java-eap-app/todo-demo-application.png)
+
+1. Check the application health endpoints. These endpoints will be used by OpenShift to verify when your pod is live and ready to receive user requests:
```

Comment:
You say 'live and ready' but the example is a liveness check, not a readiness check.